easybuild/easyblocks/i/imkl.py

```diff
@@ -434,44 +434,45 @@ def sanity_check_step(self):
 
         super(EB_imkl, self).sanity_check_step(custom_paths=custom_paths)
 
-    def make_module_req_guess(self):
+    def make_module_step(self, *args, **kwargs):
         """
-        A dictionary of possible directories to look for
+        Set paths for module load environment based on the actual installation files
         """
-        guesses = super(EB_imkl, self).make_module_req_guess()
-
         if LooseVersion(self.version) >= LooseVersion('2021'):
             compiler_subdir = os.path.join(self.get_versioned_subdir('compiler'), self.compiler_libdir)
-            pkg_config_path = [os.path.join(self.mkl_basedir, 'tools', 'pkgconfig'),
-                               os.path.join(self.mkl_basedir, 'lib', 'pkgconfig')]
+            pkg_config_path = [
+                os.path.join(self.mkl_basedir, 'tools', 'pkgconfig'),
+                os.path.join(self.mkl_basedir, 'lib', 'pkgconfig'),
+            ]
         else:
             compiler_subdir = os.path.join('lib', 'intel64')
             pkg_config_path = [os.path.join(self.mkl_basedir, 'bin', 'pkgconfig')]
-            guesses['MANPATH'] = ['man', os.path.join('man', 'en_US')]
-            guesses['MIC_LD_LIBRARY_PATH'] = [
-                os.path.join('lib', 'intel64_lin_mic'),
-                os.path.join(self.mkl_basedir, 'lib', 'mic'),
-            ]
-        library_path = [
+
+        self.module_load_environment.PATH = []
+        self.module_load_environment.LD_LIBRARY_PATH = [
             compiler_subdir,
             os.path.join(self.mkl_basedir, 'lib', 'intel64'),
         ]
```

Comment:
@lexming Are we missing this?
```suggestion
        ]
        self.module_load_environment.LIBRARY_PATH = self.module_load_environment.LD_LIBRARY_PATH

Reply:
fixed in 1658127c2